packages/select/src/Select/components/Root/Root.tsx

```diff
@@ -0,0 +1,100 @@
+import React from "react"
+import { SingleSelection } from "@react-types/shared"
+import classnames from "classnames"
+import { Label, FieldMessage } from "@kaizen/draft-form"
+import {
+  MenuTriggerConsumer,
+  MenuTriggerProvider,
+  MenuTriggerProviderContextType,
+  MenuTriggerProviderProps,
+} from "../../provider/MenuTriggerProvider"
+import {
+  SelectionConsumer,
+  SelectionProvider,
+  SelectionProviderContextType,
+} from "../../provider/SelectionProvider"
+import { ItemType } from "../../types"
+import { FloatingSelectWrapper } from "../FloatingSelectWrapper"
+import { HiddenSelectWrapper } from "../HiddenSelectWrapper/HiddenSelectWrapper"
+import rootStyles from "./rootStyles.module.scss"
+export interface RootProps extends MenuTriggerProps, SelectionProps {
+  trigger: (value?: MenuTriggerProviderContextType) => React.ReactNode
+  children: (value?: SelectionProviderContextType) => React.ReactNode // the content of the menu
+  description?: React.ReactNode
+}
+
+type MenuTriggerProps = Omit<MenuTriggerProviderProps, "children">
+
+interface SelectionProps {
+  id: string // provide A11y context for label
+  label: string // provide A11y context for listbox
+  items: ItemType[]
+  selectedKey?: SingleSelection["selectedKey"]
+  onSelectionChange?: SingleSelection["onSelectionChange"]
+}
+
+export type SelectProps = RootProps
+
+export const Root: React.VFC<RootProps> = ({
+  id,
+  trigger,
+  children,
+  isOpen,
+  defaultOpen,
+  onOpenChange,
+  label,
+  items,
+  selectedKey,
+  onSelectionChange,
+  isDisabled,
+  isFullWidth,
+  description,
+}) => {
+  const menuTriggerProps = {
+    isOpen,
+    defaultOpen,
+    onOpenChange,
+    isDisabled,
+    isFullWidth,
+    description,
+    items,
+  }
+
+  const selectionProps = {
+    id,
+    label,
+    items,
+    selectedKey,
+    onSelectionChange,
+  }
+
+  return (
+    <div>
+      <Label
+        htmlFor={id}
+        aria-label={label}
+        classNameOverride={classnames([
+          !isFullWidth && rootStyles.notFullWidth,
+        ])}
+      >
+        {label}
+      </Label>
+      <MenuTriggerProvider {...menuTriggerProps}>
+        <div className={rootStyles.container}>
+          <HiddenSelectWrapper items={items} label={label} name={id} />
+          <MenuTriggerConsumer>{trigger}</MenuTriggerConsumer>
```

Comment:
trigger (TriggerButtonBase) use the hook `useMenuTriggerContext` inside the component, you don't need to `MenuTriggerConsumer` hoc around it to make it work.

if you remove `MenuTriggerConsumer` you need to change the type `trigger: (value?: MenuTriggerProviderContextType) => React.ReactNode` to `trigger: (value?: MenuTriggerProviderContextType) => React.ReactNode`

the same can be said about `SelectionConsumer`

if you need a be more context around this we can huddle on it :)